cmd/docker/docker.go

```diff
@@ -172,9 +172,16 @@ func noArgs(cmd *cobra.Command, args []string) error {
 func main() {
 	// Set terminal emulation based on platform as required.
 	stdin, stdout, stderr := term.StdStreams()
-	logrus.SetOutput(stderr)
-
-	dockerCli := command.NewDockerCli(stdin, stdout, stderr, contentTrustEnabled(), containerizedengine.NewClient)
+	logrus.SetOutput(os.Stderr)
+
+	dockerCli := command.NewDockerCli(
+		command.WithInputStream(stdin),
```

Comment:
within the CLI code, no option should be required. (good defaults for streams, containerizedClient and content trust in the constructor). This is to make it easier for 3rd party to have (by default) the exact same behavior as the CLI itself, and use the options only to customize this behavior.